internal/controller/runner_controller.go

```diff
@@ -343,10 +355,15 @@ func (r *RunnerReconciler) fetchRunnerInstancesByNamespacedPools(instanceClient
 }
 
 func getRunnerDiff(runnerCRs, garmRunners []string) []string {
+	cache := make(map[string]struct{})
 	var diff []string
 
+	for _, runner := range garmRunners {
+		cache[runner] = struct{}{}
+	}
+
 	for _, runnerCR := range runnerCRs {
-		if !slices.Contains(garmRunners, runnerCR) {
+		if _, found := cache[runnerCR]; !found {
 			diff = append(diff, runnerCR)
 		}
 	}
```

Comment:
why is the new `cache` var needed?
doesn't do the old `if !slices.Contains(garmRunners, runnerCR) {` implementation exactly the same as the new `if _, found := cache[runnerCR]; !found {`?

I personally would stick to the funcs of the official slices package.

Reply:
this is just about optimization. Before that we had a O(n^2) runtime, as there were basically two nested for loops. Did some benchmarking on that with the benchmark package, and this solution is much more efficient because of the maps constant lookup times but of course has a bit higher memory footprint